core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -337,6 +342,33 @@ protected Integer getLivenessCheckPort() {
         }
     }
 
+    /**
+     * @return the ports on which to check if the container is ready
+     */
+    @NotNull @NonNull
+    protected Set<Integer> getLivenessCheckPorts() {
+        final Set<Integer> result = new HashSet<>();
+        if (exposedPorts.size() > 0) {
+            result.addAll(exposedPorts.stream()
```

Comment:
Can we put these blocks into private methods? Something like `getExposedPortsAsInt` and `getBoundPortsAsInt`.